bin/ream/src/main.rs

```diff
@@ -77,11 +78,21 @@ async fn main() {
                 gossipsub_config,
             };
 
-            let ream_db = ReamDB::new(config.data_dir, config.ephemeral)
+            if config.purge_db {
+                reset_db(config.data_dir.clone()).expect("Unable to delete database");
```

Comment:
We shouldn't be passing config.data_dir here. We should be passing the result of

```
let ream_dir =
            dir::setup_data_dir(APP_NAME, data_dir, ephemeral).map_err(StoreError::Io)?;
```

Refactor it out of `ReamDB::new` then pass it in both here `reset_db(`, and in ` `ReamDB::new``            
         
         
  Both those shouldn't be taking Options anymore